fastlane/Fastfile

```diff
@@ -61,6 +61,56 @@ lane :github_release do |options|
   )
 end
 
+desc "Update dependencies to latest GitHub releases and opens a PR"
+lane :open_pr_upgrading_dependencies do |options|
+  dry_run = options[:dry_run]
+  
+  new_versions = update_native_dependencies_to_latest
+  
+  repo_status = sh("git status --porcelain")
+
+  repo_clean = repo_status.empty?
+
+  if repo_clean
+    UI.success('Git status is clean, nothing to update! 💪')
+    next
+  end
+
+  new_ios_version = new_versions[:new_ios_version]
+  new_android_version = new_versions[:new_android_version]
+
+  if dry_run
+    UI.message("ℹ️  Nothing more to do since dry_run: true.")
+    UI.message("ℹ️  New iOS dependency version is #{new_ios_version}")
+    UI.message("ℹ️  New Android dependency version is #{new_android_version}")
+  else
+    branch = "update-ios-#{new_ios_version}-android-#{new_android_version}"
+
+    result = sh("git checkout -b #{branch} origin/main")
+  
+    UI.user_error!("Git couldn't change to `#{branch}`. Reason: #{result}.") if result =~ /error:(.)/
```

Comment:
I believe if the `sh` command fails it should also cause the whole lane to fail. This is mostly to make the error better looking right? (which is great! 😄)

Reply:
Oh, so you think it won't execute?

Reply:
I think so but we should probably test it. If you've tested and it does execute it, I think it's fine

Reply:
you're right, it just fails and it doesn't execute it, so we can remove these!